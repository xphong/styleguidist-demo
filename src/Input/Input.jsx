import React from 'react';
import PropTypes from 'prop-types';

export default class Input extends React.PureComponent {
  static propTypes = {
    /** Sample prop description that you can change */
    dummyProp: PropTypes.number
  };

  static defaultProps = {
    dummyProp: 123
  };

  render() {
    return (
      <p>replace me {this.props.dummyProp}</p>
    );
  }
}
