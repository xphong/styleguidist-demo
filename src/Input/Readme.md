```js
<Input dummyProp={123} />
```
