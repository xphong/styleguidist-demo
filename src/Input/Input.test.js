import React from 'react';
import { shallow } from 'enzyme';

import Input from './Input';

const createComponent = newProps => {
  const component = shallow(
    <Input {...newProps} />
  );

  return {
    component
  };
};

describe('Input component', () => {
  it('should render', () => {
    const { component } = createComponent();
    expect(component).toMatchSnapshot();
  });
});
