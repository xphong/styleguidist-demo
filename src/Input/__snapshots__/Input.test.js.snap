// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render 1`] = `
<p>
  replace me 
  123
</p>
`;
